src/devtools/client/webconsole/actions/filters.js

```diff
@@ -14,41 +14,45 @@ const {
   FILTERS,
 } = require("devtools/client/webconsole/constants");
 
-function filterTextSet(text) {
-  return {
-    type: FILTER_TEXT_SET,
-    text,
+export function filterTextSet(text) {
+  return ({ dispatch, getState }) => {
+    const filtersState = { ...getAllFilters(getState()), text }
+    return dispatch({
+      type: FILTER_TEXT_SET,
+      filtersState,
+      text,
+    });
   };
 }
 
-function filterToggle(filter) {
+export function filterToggle(filter) {
   return ({ dispatch, getState, prefsService }) => {
+    const filters = getAllFilters(getState())
+    const filtersState = { ...filters, [filter]: !filters[filter] };
```

Comment:
`{ ...filters, [filter]: !filters[filter] }` took a good second to process, but makes sense.